From 9c2c9ca2830b5ed03057f85025d9d3026da9def7 Mon Sep 17 00:00:00 2001
From: Murala S Swamy <ssmurala@habana.ai>
Date: Fri, 23 Aug 2024 15:33:03 +0300
Subject: [PATCH] [FSW-8694] fs1: set spi efuse emul in cold flops

With this patch, added support to save status of
spi/fake efuse emulation done in cold flops.
Upon any reset boot rom can retrieve and use,
only in cold reset this flops get cleared.

Change-Id: I5b6363a3d595b908b886d8359ff6468e2b4abae5
---

diff --git a/arch/fs1/arc/common.c b/arch/fs1/arc/common.c
index 2e7be7d..23ffef6 100644
--- a/arch/fs1/arc/common.c
+++ b/arch/fs1/arc/common.c
@@ -214,6 +214,19 @@
 	regmap_writel(cold_rst_data.data, IO0_H0_PSOC_BOOT(mmPSOC_BOOT_CONF_COLD_RST_REG_2));
 }
 
+void boot_rom_set_spi_efuse_read(void)
+{
+	struct fs1_cold_rst_data cold_rst_data;
+
+	cold_rst_data.data = regmap_readl(IO0_H0_PSOC_BOOT(mmPSOC_BOOT_CONF_COLD_RST_REG_2));
+	if (cold_rst_data.fuse_emulation_init_done)
+		return;
+
+	/* fake/spi efuse init successful, set cold flops */
+	cold_rst_data.fuse_emulation_init_done = 1;
+	regmap_writel(cold_rst_data.data, IO0_H0_PSOC_BOOT(mmPSOC_BOOT_CONF_COLD_RST_REG_2));
+}
+
 void boot_rom_set_tpm_extend_done(void)
 {
 	struct fs1_cold_rst_data cold_rst_data;
diff --git a/common/common.h b/common/common.h
index 5302153..3b34b57 100644
--- a/common/common.h
+++ b/common/common.h
@@ -17,6 +17,7 @@
 void boot_rom_halt(void);
 enum boot_rom_status boot_rom_read_config_fuse(void);
 void boot_rom_set_efuse_read(void);
+void boot_rom_set_spi_efuse_read(void);
 bool boot_rom_is_valid_address_range(uint64_t addr, int len);
 void boot_rom_set_tpm_extend_done(void);
 enum boot_rom_tpm_exec_stage boot_rom_get_tpm_exec_stage(void);
